File: android/src/main/kotlin/jyahann/dgis_map_kit/MapObjectUserData.kt
package jyahann.dgis_map_kit

enum class MapObjectUserDataType {
    MARKER,
    CLUSTER
}

class MapObjectUserData(type: MapObjectUserDataType, userData: Any?, layerId: String?) {
    var type: MapObjectUserDataType;
    var userData: Any?;
    var layerId: String?;

    init {
        this.type = type
        this.layerId = layerId
        this.userData = userData
    }
}

File: ios/Classes/MapObjectUserData.swift
enum MapObjectUserDataType {
    case marker;
    case cluster;
}

class MapObjectUserData {
    var type: MapObjectUserDataType;
    var userData: Any?;
    var layerId: String?;
    
    init(userDataType type: MapObjectUserDataType, userData data: Any? = nil, layerId id: String? = nil) {
        self.type = type;
        self.userData = data;
        self.layerId = id;
    }
}
